### PR TITLE
Add missing tests for temporal unit parameters parsing and fix issues

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/42118

Previously it was possible to pass invalid temporal unit parameters
- via URL
- via click behaviors

This PR fixes these issues and adds the corresponding tests.

Also found a BE issue https://github.com/metabase/metabase/issues/44684

### CODE REVIEW

File: e2e/test/scenarios/dashboard-filters/temporal-unit-parameters.cy.spec.js [hunk 1]
```diff
@@ -779,6 +848,34 @@ describe("scenarios > dashboard > temporal unit parameters", () => {
...
+        visitDashboard(dashboard.id, { params: { unit_of_time: "invalid" } });
+      });
+      filterWidget().within(() => {
+        cy.findByText(parameterDetails.name).should("be.visible");
+        cy.findByText(/invalid/i).should("not.exist");
+      });
+      getDashboardCard().findByText("Created At: Month").should("be.visible");
+    });
+
+    it("should accept temporal units outside of the allowlist if they are otherwise valid values from the url", () => {
```
Comment (kamilmielnik) [inline e2e/test/scenarios/dashboard-filters/temporal-unit-parameters.cy.spec.js:863]: My understanding is that values which aren't whitelisted should not be allowed

File: frontend/src/metabase-lib/v1/parameters/utils/parameter-parsing.ts [hunk 2]
```diff
@@ -33,14 +34,19 @@ export function getParameterValueFromQueryParams(
   return normalizeParameterValueForWidget(parsedValue, parameter);
 }
 
-function parseParameterValue(value: any, parameter: Parameter) {
+export function parseParameterValue(value: any, parameter: Parameter) {
+  const type = getParameterType(parameter);
+  if (type === "temporal-unit") {
+    const availableUnits = Lib.availableTemporalUnits();
+    return availableUnits.some(unit => unit === value) ? value : null;
```
Comment (kamilmielnik) [inline frontend/src/metabase-lib/v1/parameters/utils/parameter-parsing.ts:41]:
```suggestion
    return availableUnits.includes(value) ? value : null;
```

File: e2e/test/scenarios/dashboard-filters/temporal-unit-parameters.cy.spec.js (hunk 1 above)
Comment (ranquild) [reply e2e/test/scenarios/dashboard-filters/temporal-unit-parameters.cy.spec.js:863]: This is not the case - we discussed it with Product

File: frontend/src/metabase-lib/v1/parameters/utils/parameter-parsing.ts (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase-lib/v1/parameters/utils/parameter-parsing.ts:41]: Won't work with TS

**Review by kamilmielnik — APPROVED** [review]

File: frontend/src/metabase-lib/v1/parameters/utils/parameter-parsing.ts (hunk 2 above)
Comment (uladzimirdev) [reply frontend/src/metabase-lib/v1/parameters/utils/parameter-parsing.ts:41]: works just fine for me locally

Comment (ranquild) [reply frontend/src/metabase-lib/v1/parameters/utils/parameter-parsing.ts:41]: Oh, because `value` is `any`

**Pull request merged**